studio/stores/pgmeta/ForeignTableStore.ts

```diff
@@ -20,6 +20,33 @@ export default class ForeignTableStore extends PostgresMetaInterface<Partial<Pos
     super(rootStore, dataUrl, headers, options)
   }
 
+  // Customize ForeignTableStore fetchData method to improve request performance
+  async fetchData() {
+    const headers = { 'Content-Type': 'application/json', ...this.headers }
+    // load all foreign tables w/o columns info
+    const _url = new URL(this.url)
+    _url.searchParams.set('include_columns', 'false')
+    const url = `${_url}`
+    // load all columns
+    const urlColumns = this.url.replace('/foreign-tables', '/columns')
+    const [foreignTablesResponse, columnsResponse] = await Promise.all([
+      get(url, { headers }),
+      get(urlColumns, { headers }),
+    ])
+    if (foreignTablesResponse.error) throw foreignTablesResponse.error
+    if (columnsResponse.error) throw columnsResponse.error
+
+    // merge 2 response to create the final array
+    const foreignTables: PostgresTable[] = []
+    foreignTablesResponse.forEach((foreignTable: PostgresTable) => {
+      const columns = columnsResponse.filter((x: PostgresColumn) => x.table_id === foreignTable.id)
```

Comment:
`filter` inside for-each can get pretty slow and continuously gets slower with increasing table/columns size. I'd recommend using lodash `groupBy` before, so something like

```
 const columnsByTable = _.groupBy(columnsResponse, 'table_id')

foreignTablesResponse.forEach((foreignTable: PostgresTable) => {
   const columns = columnsResponse[foreignTable.id]

....
````

Reply:
Updated, thanks!